dags/common/ftp_service.py

```diff
@@ -0,0 +1,79 @@
+import ftplib
+import io
+import os
+import re
+import traceback
+
+from common.exceptions import DirectoryNotFoundException, NotConnectedException
+from common.utils import append_file_if_not_in_excluded_directory, walk_ftp
+from structlog import get_logger
+
+
+class FTPService:
+    def __init__(
+        self,
+        host="localhost",
+        username="airflow",
+        password="airflow",
+        port=2222,
+        dir="/upload",
+    ):
+        self.connection = None
+        self.host = host
+        self.username = username
+        self.password = password
+        self.port = port
+        self.logger = get_logger().bind(class_name=type(self).__name__)
+        self.dir = dir
+
+    def __connect(self):
+        ftp = ftplib.FTP(self.host)
+        ftp.login(self.username, self.password)
+        try:
+            ftp.dir(self.dir)
+        except FileNotFoundError:
+            raise DirectoryNotFoundException(
+                "Remote directory doesn't exist. Abort connection."
+            )
+        return ftp
+
+    def __enter__(self):
+        self.connection = self.__connect()
+        return self
+
+    def __exit__(self, exc_type, exc_value, tb):
```

Comment:
can we call it `exception_type` and `exc_value` instead (if the naming is not forced by the library)?